201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json

```diff
@@ -0,0 +1,511 @@
+                        "description": "Self hosted gateway bringing API Management to the edge"
+                    }
+                },
+                {
+                    "type": "loggers",
+                    "apiVersion": "2020-12-01",                    
+                    "name": "AppInsightsLogger",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.ApiManagement/service', variables('apim-name'))]"
+                    ],
+                    "properties": {
+                        "loggerType": "applicationInsights",
+                        "resourceId": "[resourceId('Microsoft.Insights/components', variables('app-insights-name'))]",
+                        "credentials": {
+                            "instrumentationKey": "[reference(variables('app-insights-name')).InstrumentationKey]"
+                        }
+                    }
+                },
+                {
+                    "type": "diagnostics",
+                    "apiVersion": "2020-12-01",
+                    "name": "applicationinsights",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.ApiManagement/service', variables('apim-name'))]",
+                        "[resourceId('Microsoft.ApiManagement/service/loggers', variables('apim-name'), 'AppInsightsLogger')]"
+                    ],
+                    "properties": {
+                        "alwaysLog": "allErrors",
+                        "httpCorrelationProtocol": "Legacy",
+                        "verbosity": "information",
+                        "logClientIp": true,
+                        "loggerId": "[resourceId('Microsoft.ApiManagement/service/loggers/', variables('apim-name'), 'AppInsightsLogger')]",
+                        "sampling": {
+                            "samplingType": "fixed",
+                            "percentage": 100
+                        },
+                        "frontend": {
+                            "request": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            }
+                        },
+                        "backend": {
+                            "request": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            }
+                        }
+                    }
+                },
+                {
+                    "type": "providers/diagnosticSettings",
```

Comment:
a few changes here:
1) move this to a top level resource
1) add the scope property
1) remove providers from the type

Reply:
I'm not sure that I follow on the syntax you're looking for here. I can move to top level resource, but the docs related to diagnosticSettings (https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/resource-manager-diagnostic-settings) shows needing a "type" following the pattern `resourceProvider/resourceType/providers/diagnosticSettings` and multi-segment name to appropriately associate the diagnostic setting with the correct resource. Is there an alternative syntax that is preferred here?

Reply:
Here's an example:

https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.cache/redis-premium-cluster-diagnostics/azuredeploy.json#L95-L96

TLDR; https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/scope-extension-resources?tabs=azure-cli

Reply:
Thanks for that, Brian. How to use the "scope" property was the key piece that I was missing and it makes sense now. I'll update to reflect this and suggest the doc that I linked be updated to reflect this practice as well

Reply:
Done